File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        @cache
        def go(n):
            if  n < 0 : return 0
            if n == 0 : return 1

            return go(n-1) + go(n-2)
        
        return go(n)

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int t[50];
    int climbStairs(int n) {
        memset(t,0,sizeof(t));
        t[n] = 1;
        for(int i = n - 1; i >= 0 ; i--){
            t[i] = t[i+1] + t[i+2];
        }
        return t[0];
    }
};